src/services/localStorageService.js

```diff
@@ -0,0 +1,34 @@
+export const removeFromHistory = placeId => {
+  const locationsHistory = JSON.parse(localStorage.getItem('locationsHistory')) || [];
+  const locationIndex = locationsHistory.findIndex(location => location.placeId === placeId);
+
+  if (locationIndex === -1) {
+    return;
+  }
+
+  locationsHistory.splice(locationIndex, 1);
+  localStorage.setItem('locationsHistory', JSON.stringify(locationsHistory));
+};
+
+export const addToHistory = ({ placeName, placeId, lat, lng }) => {
+  removeFromHistory(placeId);
+
+  const locationsHistory = JSON.parse(localStorage.getItem('locationsHistory')) || [];
```

Comment:
JSON parse is a bit tricky - this could throw en error - good practice is to cover parse into try/catch block